src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsGeofencingZoneMapper.java

```diff
@@ -0,0 +1,73 @@
+package org.opentripplanner.updater.vehicle_rental.datasources;
+
+import com.google.common.hash.Hashing;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.entur.gbfs.v2_2.geofencing_zones.GBFSFeature;
+import org.entur.gbfs.v2_2.geofencing_zones.GBFSGeofencingZones;
+import org.locationtech.jts.geom.Geometry;
+import org.opentripplanner.framework.geometry.GeometryUtils;
+import org.opentripplanner.framework.geometry.UnsupportedGeometryException;
+import org.opentripplanner.routing.vehicle_rental.GeofencingZone;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A mapper from the raw GBFS type into the internal model of the geofencing zones.
+ */
+class GbfsGeofencingZoneMapper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GeofencingZone.class);
+
+  private final String systemId;
+
+  public GbfsGeofencingZoneMapper(String systemId) {
+    this.systemId = systemId;
+  }
+
+  public List<GeofencingZone> mapGeofencingZone(GBFSGeofencingZones input) {
+    return input
+      .getData()
+      .getGeofencingZones()
+      .getFeatures()
+      .stream()
+      .map(this::toInternalModel)
+      .filter(Objects::nonNull)
+      .toList();
+  }
+
+  /**
+   * Convert the GBFS type to the internal model.
+   */
+  @Nullable
+  private GeofencingZone toInternalModel(GBFSFeature f) {
+    try {
+      var g = GeometryUtils.convertGeoJsonToJtsGeometry(f.getGeometry());
+      var name = Objects.requireNonNullElseGet(f.getProperties().getName(), () -> fallbackId(g));
+      var dropOffBanned = !f.getProperties().getRules().get(0).getRideAllowed();
+      var passThroughBanned = !f.getProperties().getRules().get(0).getRideThroughAllowed();
+      return new GeofencingZone(
+        new FeedScopedId(systemId, name),
+        g,
+        dropOffBanned,
+        passThroughBanned
+      );
+    } catch (UnsupportedGeometryException e) {
```

Comment:
Can you make the try-catch to be only around the single line, which can throw this type of exception?

Reply:
ae386fc4172d82c031861efdf898316d899405c4